### PR TITLE
Add query string to type definition (gatsby-config.js) in order to handle i18n

### PR DESCRIPTION
**Pull request by 8byr0**
@alexandrebodin there we go, I applied your idea that was indeed the best way to handle this problem.

I've updated README accordingly.

I've also added `queryLimit` param to this new behavior. It can now be overriden individually for each type.

### CODE REVIEW

**Comment by 8byr0** [comment]
Is the CLA signature ok? I've done but does not seem to have worked, still asking for it on this page!

File: src/fetch.js [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+module.exports = async (entityDefinition, ctx) => {
   const { apiURL, queryLimit, jwtToken, reporter } = ctx;
 
+  const { endpoint, api } = entityDefinition;
+
+  // Place global params first, so that they can be overriden by api.qs
+  const params = { _limit: queryLimit, ...api?.qs };
+
+  // Retrieve qs params if defined (or empty string instead)
+  const qs = Object.keys(params)
```
Comment (alexandrebodin) [inline src/fetch.js:13]: this isn't necessary with axios you can pass the query string as an object into the params property of the options

example:

```js
const requestOptions = {
  method: 'GET',
  url: apiBase,
  params: { _limit: 100, ...api?.qs },
  headers: {}
}

await axios(requestOptions)
```

**Review by alexandrebodin — CHANGES_REQUESTED** [review]

File: src/fetch.js (hunk 1 above)
Comment (8byr0) [reply src/fetch.js]: Right, better like this... Fixed in commit below

File: README.md [hunk 2]
```diff
@@ -58,6 +69,23 @@ You can query Document nodes created from your Strapi API like the following:
 }
 ```
 
+You can query Document nodes in chosen language
+> make sure to add `api.qs._locale` to your strapi conf in `gatsby-config.js` (see example above)
```
Comment (remidej) [inline README.md:73]: [Strapi conf](https://conf.strapi.io/) is its own thing, so I'd recommend typing "configuration" entirely

Also, I don't think this needs to be a quote (`>`), you can add this as text directly

And I would say "You can query Document nodes in *a* chosen language"

**Review by remidej — CHANGES_REQUESTED** [review]
Hello! Thanks for submitting this PR so early after the i18n release.

I tested it locally and it's working fine. I just have one concern with it. It's about these examples in the readme:

```js
// exemple fetching only english content
{
   name: `collection-name`,
   api: { qs: { _locale: 'en' } }
 },
```

It seems like when we provide an entityDefinition object, the pluralize module is disabled. So I'm afraid users might copy this config and get a 404, because the plugin will not pluralize the collection name to find the endpoint.

So I would recommend either:

* Adding the endpoint property to the readme examples
* Only disabling the pluralize module if the endpoint key is provided. Not just if there's an entityDefinition object

**Comment by 8byr0** [comment]
@remidej you were right, I've been missing pluralization if no endpoint provided, this is fixed now. (So I've not updated the example in the README, will work just fine now).
README typos fixed too

**Review by remidej — APPROVED** [review]
Thanks for the fix, it's working well now

**Comment by juliensl** [comment]
Hi guys ! 

Thank you for your work :D 

Do you have an idea for when it will be usable and published in npm ? :) 

Thank you !

**Comment by remidej** [comment]
Thanks again @8byr0 

@juliensl we'll publish this as part of a new alpha release on NPM early next week. We're just waiting for a couple more changes we'd like to add to that release

**Comment by juliensl** [comment]
So great ! Thank you @remidej and thank you too @8byr0  :smiley:

**Comment by juliensl** [comment]
Hi everybody ! :) 

Do you know when the feature will be available on npm ?
Or, do you know how can I add this commit in my plugin please ?

Thank you

**Comment by remidej** [comment]
Hello @juliensl

We just published an alpha release with this feature: https://www.npmjs.com/package/gatsby-source-strapi/v/1.0.0-alpha.2

Just know that there's a small mistake in the readme example that I'll fix tomorrow:

`filter: { locale: “en” }`
should be :arrow_down:
`filter: { locale: { eq: “en”  } }`

**Pull request merged**